pajbot/apiwrappers/seventv.py

```diff
@@ -0,0 +1,83 @@
+from pajbot.apiwrappers.base import BaseAPI
+from pajbot.apiwrappers.response_cache import ListSerializer
+from pajbot.models.emote import Emote
+
+
+class SevenTVAPI(BaseAPI):
+    def __init__(self, redis):
+        super().__init__(base_url="https://api.7tv.app/v2/", redis=redis)
+
+    @staticmethod
+    def parse_emotes(api_response_data):
+        def get_url(emote_id, size):
+            return f"https://cdn.7tv.app/emote/{emote_id}/{size}x"
+
+        emotes = []
+        for emote_data in api_response_data:
+            emote_id = emote_data["id"]
+            emotes.append(
+                Emote(
+                    code=emote_data["name"],
+                    provider="7tv",
+                    id=emote_id,
+                    urls={
+                        "1": get_url(emote_id, "1"),
+                        "2": get_url(emote_id, "2"),
+                        "3": get_url(emote_id, "3"),
+                        "4": get_url(emote_id, "4"),
+                    },
+                )
+            )
+        return emotes
+
+    def fetch_global_emotes(self):
+        """Returns a list of global 7TV emotes in the standard Emote format."""
+        query_string = """{
+            search_emotes(query: "", globalState: "only", limit: 150, pageSize: 150) {
+                id
+                name
+            }
+        }"""
+        params = {"query": query_string}
+        response = self.post("gql", json=params)
+        return self.parse_emotes(response["data"]["search_emotes"])
```

Comment:
I don't like that we don't paginate here (download multiple pages if necessary)

Reply:
I implemented the pagination by checking if we get a full page from the query.
There's  alse the `x-collection-size` header that could be checked for the total number of emotes, would require a post_response method to be added to the base api. Might be  a better approach.